src/js/plugins/plugin.url.js

```diff
@@ -145,6 +145,13 @@ BookReader.prototype.urlUpdateFragment = function() {
     return;
   }
 
+  //filtering query parameters to select only book search param (?q=foo)
+  //this needs to be updated/URL system modified if future query params are to be added
+  const queries = newQueryString.replace("?","").split("&");
+  const searchQueryIndx = queries.findIndex((el) => el.match(/^q=/));
+  const newQueryStringSearch = searchQueryIndx == -1 ? "" : queries[searchQueryIndx];
```

Comment:
Refactor this to use `URLSearchParams` instead of manual parsing, and do this inside the `else` clause below (since that's the only place it's needed).